doc/changelog/05-Ltac-language/18946-apply.rst

```diff
@@ -0,0 +1,7 @@
+- **Changed:**
+  When used with a direction :n:`->` or :n:`<-`, :tacn:`apply` now accepts an open term,
+  assuming its holes can be inferred during application, as was already the case for plain
```

Comment:
I'm not sure what the precise meaning of "assuming" is supposed to be here.
For instance if we do
~~~coq
Axiom foo : unit -> True <-> True.

Goal True.
  apply <- (foo _).
~~~
in master it fails and with this PR it succeeds leaving a shelved `unit` goal.

Reply:
Good catch. I should trust my Ltac reading skills less.

Reply:
Sure, but note that the following one succeeds on master too.
```coq
Axiom foo : unit -> True <-> True.
Goal True.
  let v := open_constr:(foo _) in apply <- v.
```

So, this pull request is not adding some new semantic behaviors, it is just removing a syntactic limitation.

If you think this will make the changelog clearer, I can stop the entry just after "open term".

Reply:
Not sure how that matters, you can do regular `apply` with a ltac-bound open_constr too.

Reply:
Everybody agrees, and if you know how to do that, feel free to implement it.

Reply:
I made another attempt at https://github.com/andres-erbsen/coq/tree/fix-18177; it passes both tests so far

Reply:
Thank you for working on it. What are we supposed to review and test, this PR or @andres-erbsen's branch?